domain/externalcontroller/modelmigration/export.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelmigration
+
+import (
+	"context"
+
+	"github.com/juju/description/v4"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/crossmodel"
+	"github.com/juju/juju/core/modelmigration"
+	"github.com/juju/juju/domain"
+	"github.com/juju/juju/domain/externalcontroller"
+	"github.com/juju/juju/domain/externalcontroller/service"
+	"github.com/juju/juju/domain/externalcontroller/state"
+)
+
+// RegisterExport registers the export operations with the given coordinator.
+func RegisterExport(coordinator Coordinator) {
+	coordinator.Add(&exportOperation{})
+}
+
+type ExportService interface {
+	ControllerForModel(ctx context.Context, modelUUID string) (*crossmodel.ControllerInfo, error)
+	ModelsForController(ctx context.Context, controllerUUID string) ([]string, error)
+}
+
+// exportOperation describes a way to execute a migration for
+// exporting external controller s.
```

Comment:
Space snuck in there after "controller".

Reply:
fixed